doc/Programming-guide/Key-platform-features/Endpoint-groups/index.md

```diff
@@ -6,166 +6,308 @@ nav: /:path/Programming-guide/Key-platform-features/Endpoint-groups
-Endpoint group profile filters are predicate expressions which define characteristics of group members (endpoints). These filters are executed against the endpoint profile to figure out whether or not the endpoint belongs to the group.
+* TOC
+{:toc}
 
-### NOTE ###
+Kaa provides a mechanism for endpoints aggregation within the application that is based on groups. Grouping endpoints enables you to activate specific configuration parameters, 
+control access to notification topics, etc.
+This guide will familiarize you with the basic concepts of designing endpoint groups. It is assumed that you have either set up a Kaa Sandbox, or a fully-blown Kaa cluster 
+already and that you have created a tenant and an application in Kaa. It is also strongly recommended to review 
+[Endpoint profiles]({{root_url}}Programming-guide/Key-platform-features/Endpoint-profiles/) before you proceed with this one. 
 
-	Different profile schema versions may require separate profile filters due to the schema structural differences. 
-	In case a group has no filter assigned for a specific profile schema version, the group does not apply to the endpoints that use the profile of this schema version.
+Endpoint profile is a virtual identity or "passport" of the endpoint. By filtering against the data in the profiles, endpoints may be aggregated into independently managed groups.
+Kaa allows for aggregating endpoints related to the same application into endpoint groups. The _endpoint group_ represents an independent managed entity which is 
+defined by the profile filters assigned to the group. Those endpoints whose profiles match the profile filters of the specific endpoint group become automatically 
+registered as members of this group. There is no restriction for endpoints on having membership in a number of groups at a time.
 
-Every endpoint group within the application has a unique weight value assigned to it. The weight value is used to resolve conflicts among endpoint groups: in general, the larger the weight, the higher the group priority.
-Every Kaa application comes with the default, non-user-editable group "all", with the weight value 0. The profile filter of this group is automatically set to "true" for every profile schema version in the system. As a result, the "all" group contains every endpoint registered in the application. The "all" group is used to define the default configuration, default notification topics access list, and for some other special functions.
+Endpoint group profile filters are predicate expressions which define characteristics of group members (endpoints). These filters are executed against the endpoint 
+profile to figure out whether or not the endpoint belongs to the group
```

Comment:
... to figure out whether or not the endpoint belongs to the group**.**('dot' is missing)